### ISSUE TITLE
CPTRangePlot - Area Fill - Not filling with color

### ISSUE THREAD

**Issue by mbknieri** [core-plot/core-plot#326]
I need to plot a graph with 2 scatterplots (min and max) with the area in between the scatterplots being filled with color. I tried to implement a workaround using CPTRangePlot and setting the area fill property to be filled with a color, but unfortunately, this is not working. The area is not filling with any color. This is how I am setting the area fill:

    CPTRangePlot *rangePlot = [[CPTRangePlot alloc] init]; 
    rangePlot.barLineStyle = nil; //I set this to nil so that the bar lines do not show
    rangePlot.dataSource = dataSource;   
    CPTColor *color = [CPTColor colorWithComponentRed:0 green:212.0f/255.0f blue:213.0f/255.0f alpha:0.1f];
    CPTFill *areaFill = [[CPTFill alloc] initWithColor:color];
    rangePlot.areaFill = areaFill;

When I set the areaFill, nothing happens. Is this a bug or is there another property in CPTRangePlot that needs to be set in order to get this to work?

NOTE: I am running against IOS 10.2, Xcode 8.2.1, using CorePlot version 1.6 (although, I have tried upgrading CorePlot to version 2.2 and I found the same issue exists in this version as well)

**Comment by eskroch** [comment]
As a test, set the `areaBorderLineStyle` to outline the filled area. If that doesn't work either, check the data provided by the datasource to make sure it falls within the plot ranges defined by the plot space. Also note that a range plot requires six data fields (x, y, high, low, left, and right). Return `NAN` or `nil` for any that you don't need, e.g., left and right.

**Comment by mbknieri** [comment]
I tried setting the areaBorderLineStyle and nothing happens, there is still no area fill. The six data fields are set. If I set the range plot's barLineStyle property, I see the bars appear in between the two plot lines, but setting the areaFill property does nothing.

**Comment by eskroch** [comment]
Can you post your setup and datasource code, some sample data, and a screenshot of what you're seeing?

**Comment by mbknieri** [comment]
I started stepping through code into the CPTRangePlot class functions where the area fill is being drawn and comparing that to the Plot Gallery sample app. I noticed that when I got to the renderAsVectorInContext method, I noticed that all properties in the CGPointError viewpoints was returning NAN after the viewpoints are calculated, which is interesting because my datasource method numberForPlot is returning values for each coordinate. Also, when I step into reloadPlotDataInIndexRange the newXValues, etc. are being set correctly and it appears as though these values are being cached successfully. I have a hunch that the data is either not being properly cached or the cached data is not being properly parsed when it comes time to draw the area fill. 

I can't post my code but I can try to give you a general sense of how everything is set up.

I have a class that is a CPTPlotDataDelegate and implements numberOfRecordsForPlot and numberForPlot:field:recordIndex. The numberForPlot method is returning an NSNumber (instead of nullable id like in the sample app). There is a separate object used to create the range plot and add it to the graph. The range plot's datasource property is being set to the CPTPlotDataDelegate class. When I set the areaFill property to a CPTFill, nothing happens. There is no fill in the graph. However, when I set the barLineStyle, bars are drawn correctly in the range plot. 

Is my thinking as to what the issue may be correct or at least in the right direction? Is there a way I need to package my data when passed to numberForPlot for it to be retrieved from the cache correctly (ex. need to be NSNumber from double)?

Let me know if theres additional info you need. Thanks for your help.

**Comment by mbknieri** [comment]
I think I figured it out!

There are some NaN values in my viewpoints array, the first 20 or so are NaN values. Since both x and y are nan for the first point, CGPathMoveToPoint is not being called (CPTRangePlot:666). This results an error with CGPathAddLineToPoint, since no there is no current point in the path, resulting in nothing being drawn. If I set the start index to the first value in my viewpoints array that is not nan, then setting the areaFill property works!

I need to look at my data and see why nan is being passed in so that the areaFill will work without my hack. I think I am all set so I will close this issue. Thank you for your help.
